src/plugins/prefs/prefs.c

```diff
@@ -0,0 +1,226 @@
+	snprintf (prefKey, keyLen + 1, "%s", sPtr);
+	char * tPtr = strtok (prefKey, ".");
+	if (tPtr) keyAddBaseName (key, tPtr);
+	while ((tPtr = strtok (NULL, ".")) != NULL)
+	{
+		keyAddBaseName (key, tPtr);
+	}
+	elektraFree (prefKey);
+	sPtr = cPtr + 2;
+	ePtr = strrchr (sPtr, ')');
+	*ePtr = '\0';
+	size_t argLen = ePtr - sPtr;
+	char * prefArg = elektraMalloc (argLen + 1);
+	snprintf (prefArg, argLen + 1, "%s", sPtr);
+	if (!strcmp (prefArg, "true") || !(strcmp (prefArg, "false")))
+	{
+		keySetMeta (key, "pref/type", "boolean");
+		keySetString (key, prefArg);
+	}
+	else if (prefArg[0] == '"' && prefArg[strlen (prefArg) - 1] == '"')
+	{
+		// TODO: else if list
+		keySetMeta (key, "pref/type", "string");
+		*prefArg = '\0';
+		*(prefArg + (strlen (prefArg + 1))) = '\0';
+		keySetString (key, (prefArg + 1));
+	}
+	else
+	{
+		keySetMeta (key, "pref/type", "integer");
+		keySetString (key, prefArg);
+	}
+	elektraFree (prefArg);
+	elektraFree (localString);
+	return key;
+}
+
+static Key * varToKey (Key * parentKey, DataType d, const char * string)
+{
+	Key * key = keyNew (keyString (parentKey), KEY_END);
+	keyAddBaseName (parentKey, "variables");
+	return key;
+}
+
+int elektraPrefsGet (Plugin * handle ELEKTRA_UNUSED, KeySet * returned ELEKTRA_UNUSED, Key * parentKey ELEKTRA_UNUSED)
+{
+	if (!elektraStrCmp (keyName (parentKey), "system/elektra/modules/prefs"))
+	{
+		KeySet * contract =
+			ksNew (30, keyNew ("system/elektra/modules/prefs", KEY_VALUE, "prefs plugin waits for your orders", KEY_END),
+			       keyNew ("system/elektra/modules/prefs/exports", KEY_END),
+			       keyNew ("system/elektra/modules/prefs/exports/open", KEY_FUNC, elektraPrefsOpen, KEY_END),
+			       keyNew ("system/elektra/modules/prefs/exports/close", KEY_FUNC, elektraPrefsClose, KEY_END),
+			       keyNew ("system/elektra/modules/prefs/exports/get", KEY_FUNC, elektraPrefsGet, KEY_END),
+			       keyNew ("system/elektra/modules/prefs/exports/set", KEY_FUNC, elektraPrefsSet, KEY_END),
+			       keyNew ("system/elektra/modules/prefs/exports/error", KEY_FUNC, elektraPrefsError, KEY_END),
+			       keyNew ("system/elektra/modules/prefs/exports/checkconf", KEY_FUNC, elektraPrefsCheckConfig, KEY_END),
+#include ELEKTRA_README (prefs)
+			       keyNew ("system/elektra/modules/prefs/infos/version", KEY_VALUE, PLUGINVERSION, KEY_END), KS_END);
+		ksAppend (returned, contract);
+		ksDel (contract);
+
+		return 1; // success
+	}
+	// get all keys
+	const char * fileName = keyString (parentKey);
+	FILE * fp = fopen (fileName, "r");
+	int len = 1024;
+	char * buffer = elektraMalloc (len * sizeof (char));
+	Key * key;
+	Key * sectionKey = keyNew (keyName (parentKey), KEY_END);
+	keySetBinary (sectionKey, 0, 0);
+	keyAddBaseName (sectionKey, "preferences");
+	ksAppendKey (returned, keyDup (sectionKey));
+	keySetBaseName (sectionKey, "variables");
+	ksAppendKey (returned, keyDup (sectionKey));
+	keySetBaseName (sectionKey, "functions");
```

Comment:
Do we really need to capture variables and functions as such?

What about taking everything which is not related to pref as comment?